### PR TITLE
sign in and sign out 

### PR DESCRIPTION
**Pull request by kouch-chivorn**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'users/show'
```
Comment (chuanhtuan) [inline config/routes.rb]: remove

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -4,7 +4,25 @@
...
+            <li id="fat-menu" class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+                <%= current_user.name %> <b class="caret"></b>
+              </a>
+              <ul class="dropdown-menu">
+                <li><%= link_to "Profile", users_show_path %></li>
+                <li><%= link_to "Settings", edit_user_registration_path %></li>
+                <li class="divider"></li>
+                <li>
+                  <%= link_to "Sign out", destroy_user_session_path, method: "delete" %>
```
Comment (chuanhtuan) [inline app/views/layouts/_header.html.erb]: :delete

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,23 @@ class ApplicationController < ActionController::Base
...
+
+  before_action :configure_devise_permitted_parameters, if: :devise_controller?
+  before_action :authenticate_user!
+
+  protected
+
+  def configure_devise_permitted_parameters
+    registration_params = [:name, :email, :password, :password_confirmation]
+
+    if params[:action] == 'update'
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: use `"` instead of `'`

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -2,4 +2,23 @@ class ApplicationController < ActionController::Base
...
+      devise_parameter_sanitizer.for(:account_update) { 
+        |u| u.permit(registration_params << :current_password)
+      }
+    elsif params[:action] == 'create'
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: use `"` instead of `'`

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by(id: params[:id])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: spec/controllers/users_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require 'spec_helper'
```
Comment (dieunb) [inline spec/controllers/users_controller_spec.rb]: remove it, spec test is later

File: spec/helpers/users_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
```
Comment (dieunb) [inline spec/helpers/users_helper_spec.rb]: remove it, spec test is later

File: spec/views/users/show.html.erb_spec.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
```
Comment (dieunb) [inline spec/views/users/show.html.erb_spec.rb]: remove it, spec test is later

File: spec/helpers/users_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Specs in this file have access to a helper object that includes
```
Comment (dieunb) [inline spec/helpers/users_helper_spec.rb]: remove this file

File: spec/views/users/show.html.erb_spec.rb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+describe "users/show.html.erb" do
```
Comment (dieunb) [inline spec/views/users/show.html.erb_spec.rb]: remove this file

File: app/models/question.rb [hunk 11]
```diff
@@ -1,3 +1,10 @@
 class Question < ActiveRecord::Base
   belongs_to :category
+  has_many :answers
+  has_many :test_questions
+
+  accepts_nested_attributes_for :answers
```
Comment (dieunb) [inline app/models/question.rb]: you should add `allow_destroy` and `reject_if` option to use some features with its

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+      redirect_to root_path, alert: "Not found user with #{params[:id]}" unless @user
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: indent

**Pull request merged**